src/components/sale/SaleCollectionAdminCard.tsx

```
@@ -341,4 +348,68 @@ const SaleCollectionAdminCard: React.VFC<{
   )
 }
 
+const DiscountCode: React.VFC<{ type: string; target: string }> = ({ type, target }) => {
+  const apolloClient = useApolloClient()
+  const [code, setCode] = useState('')
+
+  const getDiscountCode = async ({ type, target }: { type: string; target: string }) => {
+    switch (type) {
+      case 'Coupon':
+        const { data: coupon } = await apolloClient.query<
+          hasura.GET_COUPON_CODE_BY_COUPON,
+          hasura.GET_COUPON_CODE_BY_COUPONVariables
+        >({
+          query: GET_COUPON_CODE_BY_COUPON,
+          variables: {
+            id: target,
+          },
+          fetchPolicy: 'no-cache',
+        })
+        return coupon.coupon_by_pk?.coupon_code.code || ''
```

it should throw an error here if the code is empty

---

throw error or 
return empty message something like 無折扣碼 ？

---

To reduce unnecessary requests, I think we should only send requests when permission is allowed (ex: only app-owner has permission here)
if we get nothing from the response then throw an error